gratipay/billing/payday.py

```diff
@@ -706,6 +707,57 @@ def end(self):
         """, default=NoPayday).replace(tzinfo=aspen.utils.utc)
 
 
+    def notify_participants(self):
+        ts_start, ts_end = self.ts_start, self.ts_end
+        exchanges = self.db.all("""
+            SELECT amount, fee, note, status, p.*::participants AS participant
+              FROM exchanges e
+              JOIN participants p ON e.participant = p.username
+             WHERE "timestamp" >= %(ts_start)s
+               AND "timestamp" < %(ts_end)s
+               AND amount > 0
+               AND p.notify_charge > 0
+        """, locals())
+        for e in exchanges:
+            if e.status not in ('failed', 'succeeded'):
+                log('exchange %s has an unexpected status: %s' % (e.id, e.status))
+                continue
+            i = 1 if e.status == 'failed' else 2
+            p = e.participant
+            if p.notify_charge & i == 0:
```

Comment:
+1 on that. Either not use them at all, or use abstraction so that we don't have to work with bits directly.